include/targets/cuda/FFT_Plans.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include <quda_internal.h>
+#include <quda_matrix.h>
+#include <cufft.h>
+
+using FFTPlanHandle = cufftHandle;
+#define FFT_FORWARD     CUFFT_FORWARD
+#define FFT_INVERSE     CUFFT_INVERSE
+
+#ifndef GPU_GAUGE_ALG
+
+#define CUFFT_SAFE_CALL(call)
+
+inline void ApplyFFT(FFTPlanHandle &, float2 *, float2 *, int)
+{
+  errorQuda("CPU_GAUGE_ALG is disabled so FFTs are also disabled");
+}
+
+inline void ApplyFFT(FFTPlanHandle &, double2 *, double2 *, int)
+{
+  errorQuda("CPU_GAUGE_ALG is disabled so FFTs are also disabled");
+}
+
+inline void SetPlanFFTMany(FFTPlanHandle &, int4, int, QudaPrecision)
+{
+  errorQuda("CPU_GAUGE_ALG is disabled so FFTs are also disabled");
+}
+
+inline void SetPlanFFT2DMany(FFTPlanHandle &, int4, int, QudaPrecision)
+{
+  errorQuda("CPU_GAUGE_ALG is disabled so FFTs are also disabled");
+}
+
+inline void FFTDestroyPlan(FFTPlanHandle &)
+{
+  errorQuda("CPU_GAUGE_ALG is disabled so FFTs are also disabled");
+}
+
+#else
+
+/*-------------------------------------------------------------------------------*/
+#define CUFFT_SAFE_CALL( call) {                                      \
+    cufftResult err = call;                                         \
+    if ( CUFFT_SUCCESS != err ) {                                     \
+      fprintf(stderr, "CUFFT error in file '%s' in line %i.\n",   \
```

Comment:
should we switch to a QUDA specific error handling here

Reply:
b798f7aba